src/core/cli/lifecycle/deps.go

```
@@ -0,0 +1,294 @@
+	if len(agents) == 0 {
+		return nil
+	}
+	path := depsFileFor(service, group)
+	return withLifecycleLock(func() error {
+		existing, err := readDeps(path)
+		if err != nil {
+			return fmt.Errorf("lifecycle: read deps %s: %w", path, err)
+		}
+		if len(existing) == 0 {
+			// File missing or already empty — try to remove the empty file too.
+			os.Remove(path)
```

golangci-lint (2.11.4)

[error] Error return value of `os.Remove` is not checked

(errcheck)